plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/ButtonLink.tsx

```diff
@@ -20,10 +20,15 @@ const ButtonLink: React.FC<BaseFieldProps<"buttonLink">> = ({ value, editMode, o
 
   const handleChangeButtonLink = useCallback(
     (e: React.ChangeEvent<HTMLInputElement>) => {
-      setLink(e.currentTarget.value);
+      let url = e.currentTarget.value;
+      const prefix = "http://";
+      if (!url.match(/^[a-zA-Z]+:\/\//)) {
+        url = prefix + url;
```

Comment:
HAve you tested this? Also, most websites are https now, right? Perhaps https:// is better to attach. Though I am a little unsure this is good/needed. I feel like its prone to unexpected behavour

Reply:
if I didn't add ths validation when I set the link to "www.google.com " ,  causes the browser to assume a relative path and append localhost to the url , I tested that and it works fine

Reply:
I added validation for the default value also

Reply:
c9e5706fc4bb3535ff518c320d648fb4f2ea4b1b

Reply:
I think it is best to just do the regex check on the link in the onClick with the window.open.
We shouldn't be changing the saved value, because the user would get confused, and we only really need the https:// attached before opening the page.

Reply:
done & tested